### ISSUE TITLE
Regularization: GBM / XGB / LGBM / Catboost

### ISSUE THREAD

**Issue by choshin84** [choshin84/learning_memo#5]
# Tweet summary
Boosting uses tree structure setup as regularization, i.e. pruning, etc, but XGB / LGBM / Catboost has L1/L2 regularization accommodated in loss function with weight factor.

# Useful link
https://towardsdatascience.com/boosting-algorithm-xgboost-4d9ec0207d
XGB: https://xgboost.readthedocs.io/en/latest/parameter.html
LGBM: https://lightgbm.readthedocs.io/en/latest/Python-API.html
Catboost: https://catboost.ai/docs/concepts/parameter-tuning.html
